### ISSUE TITLE
calling the getXpath() method asyncronously

### ISSUE THREAD

**Issue by hyouinKyouma** [trembacz/xpath-finder#11]
I line 22 of inspect.js we are calling the getXpath method, but I want to modify the xpath for which I am calling it inside another method that I have added, lets call this method as 'extraMethod()', this method takes an object from storage.local, hence return a promise. I call the getXpath method inside this method then call this method in line 22 with await but the whole thing breaks.

What can be done in this case.

//the extra method
 getLocalStorage(el){
       const xp = this.getXPath(e)
      try{
        const storage = await browser.storage.local.get()
        const newXpath = xp + storage.attVal
        return newXpath
      }catch(err){
        return err
      }
    }

//calling the method instead of getXpath
this.XPath =  getLocalStorage(e.target);

**Comment by trembacz** [comment]
Hi @hyouinKyouma, 

Could you let me know what you expect? From what I understand you've forked my repository and do some changes which is great but looks like I don't need those changes here. If you need help with your code, please send me a link to repo and I will check what I can do to help you fix the problem. 

Thanks,
Tomek

**Comment by hyouinKyouma** [comment]
Hi Tomek

I am working on making a tool for generating relative xpath, that can be
used for scraping, for which I have added some code to the getXpath method,
and I have added two more methods 'checkForClass', 'getLocalStorage'. Now I
am concatenating the value of storage.local and return value of 'getXpath'
method inside 'getLocalStorage'. Now 'getLocalStorage' method returns a
promise.
But the value is not shown in the bottom panel.

Though, now I have solved that problem. I was missing the async at the very
top with getData() function.

here is the link to the repo: https://github.com/hyouinKyouma/xpath-finder.

Thanks
Adarsh
